pkg/network/usm/ebpf_javatls.go

```diff
@@ -146,6 +149,17 @@ func (p *JavaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPa
 	return []manager.ProbeIdentificationPair{{EBPFFuncName: "kprobe__do_vfs_ioctl"}}
 }
 
+// isJavaProcess checks if the given PID comm's name is java.
+// The method is much faster and efficient that using process.NewProcess(pid).Name().
+func isJavaProcess(pid int) bool {
+	content, err := os.ReadFile(filepath.Join(util.GetProcRoot(), strconv.Itoa(pid), "comm"))
+	if err != nil {
+		return false
+	}
+
+	return bytes.HasPrefix(content, javaProcessName)
```

Comment:
would be better to hook only on "java" string
`strings.Trim(string(content)) == "java"`
we don't want to hook on `javac`

Reply:
Can we hook on javac?
Pedro is right - https://github.com/DataDog/datadog-agent/pull/17129#discussion_r1194016261

Reply:
done, sort of